u4-databases/features/uni-controller.js

```diff
@@ -0,0 +1,54 @@
+const { db } = require('../_services/firebase-initialized')
+
+const iterationList = async (req, res) => {
+  const query = db.collection('iterations').get()
+  const iterations = (await query).docs.map(doc => doc.data())
+
+  res.render('uni-iteration-List', { iterations })
+}
+
+const iterationDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('iterations').doc(code).get()
+  const iteration = (await query).data()
+  console.log(iteration)
+
+  res.render('uni-iteration-details', { iteration })
+}
+
+const studentsDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('students').doc(code).get()
+  const student = (await query).data()
+
+  res.render('uni-student-details', { student })
+}
+
+const courseLists = async (req, res) => {
+  const query = db.collection('iterations')
+    .where('year', '==', 2021)
+    .where('semester', '==', 1)
+    .get()
+  const courses = (await query).docs.map(doc => doc.data())
+
+  res.render('uni-course-list', { courses })
+}
+
+const courseDetails = async (req, res) => {
+  const iterationCode = req.params.code
+  const query = db.collection('iterations')
+    .where('courseCode', '==', iterationCode)
```

Comment:
ชื่อตัวแปรของคุณทำให้สับสนมาก! นี่ไม่ใช่ `iterationCode' (เช่น: `231370.20s1`) มันคือ `รหัสหลักสูตร` (เช่น: `231370`)
Your variable name is very confusing! This isn't an iterationCode (ex: `231370.20s1`). It is a courseCode (ex: `231370`).